contentful/apps
# Integ 269 graph jitter bug and error flashing before load

## Purpose
This PR fixes two issues: 
1. When the analytics chart is mounting, there is a weird stuttering that occurs, that is very uncomfortable to look at. 😅 
2. An incorrect date was being provided at the end of the runReports data. (Feb 21st instead of March 21st for example). 

## Approach
1. To solve the first issue, despite diving head first into some animation adjusting, this bug actually just ended up being an issue of the chart not properly having a height and width defined, so on resize it would flicker until it properly inherited parent height and width. Therefore I added a height and width definition to its parent. This also stabilized the container itself, so rendering the warning note instead of the chart does not cause a lot of shifting around. 
2. The incorrect date was being added because the date mapping occurring on the API side was incorrectly using 'day' as a way to compare if the data already exists. So Feb 21st would be a match to March 21st. I changed this conditional to use `valueOf()` to appropriately determine the equality of dates. 

## How to test
Pull down this code and render the GA4 app in the sidebar of an entry. Ideally, notice on render and when changing the date range, there is no flickering of the chart! Additionally, the dates provided should also be an accurate representation of the date range selected. 


Video showing no jitter bug: https://www.loom.com/share/679f315fd9604988a03b165b57007e7c
